core/src/main/java/org/elasticsearch/action/ActionModule.java

```diff
@@ -357,6 +469,142 @@ public void register(ActionHandler<?, ?> handler) {
         return unmodifiableList(filters);
     }
 
+    private Set<Class<? extends RestHandler>> setupRestHandlers() {
+        ExtensionPoint.ClassSet<RestHandler> handlers = new ExtensionPoint.ClassSet<>("rest_handler", RestHandler.class);
```

Comment:
I think we should just use Set? The ExtentionPoint class was added at a time we thought that would be the new plugin model. But I don't think we should use it anymore.